### PR TITLE
Make sure that `PhpStormStubsSourceStubber` updates constants based on current runtime environment, when outdated or not defined in stubs

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

**Comment by Ocramius** [comment]
I don't understand this patch: can the commit description be expanded a bit? :sweat_smile:

**Comment by kukulich** [comment]
Stubs are often outdated or constants have different values on different systems, see comment for method `updateConstantValue` (the comment was already there - it just moved now)

So we already update the constant value based on the real value (based on `constant()` call).

The current version only updates constants in stubs defined by `define()` call. The patch fixes it so all types of constants are updated - class constants and constants defined by `const` keyword too.

I've spotted this when preparing PHP 8.1 version - some constants change in PHP 8.1.

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php [hunk 1]
```diff
@@ -587,4 +589,45 @@ public function testCaseSensitiveConstant(string $constantName): void
...
+     */
+    public function testUpdateConstantValue(): void
+    {
+        require __DIR__ . '/../../Fixture/FakeConstants.php';
+
+        $sourceStubber = new PhpStormStubsSourceStubber(BetterReflectionSingleton::instance()->phpParser());
+
+        $stubberReflection = new CoreReflectionClass($sourceStubber);
+
+        $stubDirectoryReflection = $stubberReflection->getProperty('stubsDirectory');
```
Comment (Ocramius) [inline test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php:604]: Hacky :P

**Review by Ocramius — APPROVED** [review]
LGTM :ship:

File: test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php (hunk 1 above)
Comment (kukulich) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php:604]: Yes, two hours of creating this test :D

Comment (Ocramius) [reply test/unit/SourceLocator/SourceStubber/PhpStormStubsSourceStubberTest.php:604]: I know the feeling - having one of those come up for me too, on a client project :-\

**Pull request merged**